### ISSUE TITLE
Cypress update instructions given say to update with npm, even if you're using yarn

### ISSUE THREAD

**Issue by silbinarywolf** [cypress-io/cypress#2871]
### Current behavior:
When I open Cypress and there's a new update, the update instructions tell me to use `npm install --save-dev cypress@3.1.2 ` rather than saying `yarn add --dev cypress@3.1.2`.

### Desired behavior:
If a `yarn.lock` file is detected instead of a `package-lock.json` file, the message should given you the yarn instructions.

The location of the update code is found at:
- `packages/desktop-gui/src/update/update-banner.jsx`

The location of the code that tests the update banner:
- `integration/update_banner_spec.coffee`

### Steps to reproduce:
I assume you should be able install Cypress 3.1.1, run it with ``cypress open` and see the update button / with instructions.

### Versions
- 3.1.1

### Operating System
Windows 10

**Comment by louise-hayes** [comment]
Could I take this issue?

**Comment by flotwig** [comment]
@louise-hayes Absolutely!

**Comment by louise-hayes** [comment]
Have reproduced error as per notes above, on MAC Sierra 10.12.6
![screen shot 2019-02-08 at 14 16 22](https://user-images.githubusercontent.com/29293985/52500774-c4f0de80-2bac-11e9-8af1-9aef5afd11c9.png)

**Comment by louise-hayes** [comment]
Hi!
I have added code to check for yarn.lock but the file check is always 'null', because the yarn.lock is not accessible to cypress during the test run phase.

Below are the screen shots, to show my progress. I attached the folders and file that cypress sees when running the tests, and also the code I have changed on my repo.

If you agree that the file yarn.lock is not accessible, as I believe, and we cannot check for it in the way I am trying:
Would it be a good idea  / possible to change the code in another location, during the build, to set a flag/marker which would check and show if the yarn file exists - So that this yarn file is accesible.

I am happy to do stick with this and find a solution if you think it is the right way. If anyone has any ideas how I can make this marker flag or any feedback that would be great,

Thanks!


![screen shot 2019-02-19 at 22 36 05](https://user-images.githubusercontent.com/29293985/53064850-44c55580-3497-11e9-8467-56e1fa5d1bf5.png)

![screen shot 2019-02-19 at 22 47 16](https://user-images.githubusercontent.com/29293985/53065120-5b1fe100-3498-11e9-87d7-275d1ef9a9f5.png)

![screen shot 2019-02-19 at 22 47 57](https://user-images.githubusercontent.com/29293985/53065156-75f25580-3498-11e9-8d34-83350b6f25b0.png)

**Comment by chrisbreiding** [comment]
The `desktop-gui` package is similar to a web app in a browser and doesn't have access to the user's file system. For that, it utilizes the [`server` package](https://github.com/cypress-io/cypress/tree/develop/packages/server). 

For the update check, the `desktop-gui` sends an `updater:check` message, which is [received here in the `server`](https://github.com/cypress-io/cypress/blob/2333d04a54acfd2d89d9d53cc60a49d46228ceef/packages/server/lib/gui/events.coffee#L131) and calls into [`Updater.check` here](https://github.com/cypress-io/cypress/blob/2333d04a54acfd2d89d9d53cc60a49d46228ceef/packages/server/lib/updater.coffee#L82). One possible solution is to check for the existence of `yarn.lock` in `Updater.check` and return with an `isYarn: true|false` property along with the new version. 

Honestly, it might be better to just show both options in the dialog instead of trying to determine if the user is using yarn:

```
To update Cypress:

1. Quit this app

2. If using npm, run `npm install --save-dev cypress@3.1.5`
   
   If using yarn, run `yarn add cypress@3.1.5`

3. Run ...
```

**Comment by louise-hayes** [comment]
Thanks I will implement option B, add a second message for yarn and then push that to repo for now.

**Comment by jennifer-shehane** [comment]
Released in [3.2.0](https://on.cypress.io/changelog#3-2-0)

**Comment by jennifer-shehane** [comment]
This issue will be closed to further comment as the exact issue here was resolved and tested. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
